frontend/pages/hosts/details/DeviceUserPage/EnrollMdmModal/EnrollMdmModal.tsx

```diff
@@ -11,17 +11,19 @@ import Spinner from "components/Spinner";
 
 import mdmAPI from "services/entities/mdm";
 
-export interface IInfoModalProps {
+interface IEnrollMdmModalProps {
+  mdmEnrollmentType: "manual" | "auto";
   onCancel: () => void;
-  token: string;
+  token?: string;
 }
 
-const baseClass = "manual-enroll-mdm-modal";
+const baseClass = "enroll-mdm-modal";
 
-const ManualEnrollMdmModal = ({
+const EnrollMdmModal = ({
+  mdmEnrollmentType,
   onCancel,
-  token,
-}: IInfoModalProps): JSX.Element => {
+  token = "",
```

Comment:
why is this optional? Don't you need this to make the API call correctly? I think we should make this required, or if its truly optional than handle the case where it is undefined.

Reply:
> why is this optional?

Because in the Automatic case there is no API call

> if its truly optional than handle the case where it is undefined.

okay

Reply:
ah i see ok makes sense